pkg/controller/machine/controller.go

```diff
@@ -422,21 +422,23 @@ func isInvalidMachineConfigurationError(err error) bool {
 func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, errorMessage string) error {
 	if stringPointerDeref(machine.Status.Phase) != phase {
 		klog.V(3).Infof("%v: going into phase %q", machine.GetName(), phase)
+		// Patch annotations before updating status
+		if phase == phaseFailed {
+			err := r.patchFailedMachineInstanceAnnotation(machine)
+			if err != nil {
+				klog.Errorf("Failed to update machine %q: %v", machine.GetName(), err)
+				return err
+			}
+		}
+
+		// Patch status
 		baseToPatch := client.MergeFrom(machine.DeepCopy())
 		machine.Status.Phase = &phase
 		machine.Status.ErrorMessage = nil
 		now := metav1.Now()
 		machine.Status.LastUpdated = &now
 		if phase == phaseFailed && errorMessage != "" {
 			machine.Status.ErrorMessage = &errorMessage
-			if machine.Annotations == nil {
-				machine.Annotations = map[string]string{}
-			}
-			machine.Annotations[MachineInstanceStateAnnotationName] = unknownInstanceState
-			if err := r.Client.Patch(context.Background(), machine, baseToPatch); err != nil {
```

Comment:
nit:
If we dropped 435 to 439, wouldn't it just work? That would be easier to read from my pov but that's subjective.

Reply:
oh ignore it. I thought we were doing two calls tp patch status, but one is for the object.